### PR TITLE
Bump dcos-diagnostics

### PR DESCRIPTION
**Pull request by janisz**
## High-level description

Enable optional files in diagnostics-bundle

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-4949](https://jira.mesosphere.com/browse/DCOS_OSS-4949)     Enable optional files in diagnostics-bundle

## Related tickets (optional)

Other tickets related to this change:

  - [DCOS_OSS-<number>](https://jira.mesosphere.com/browse/DCOS_OSS-<number>) Foo the Bar so it stops Bazzing.

## Checklist for all PRs

  - [ ] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): https://github.com/dcos/dcos-diagnostics/compare/096c5771987e2f489c9ba0369b080f1bb99bce03..f5e21295900ae290bbd4be41add2b707d5a7733e
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by janisz** [comment]
@mesosphere-mergebot label Ready For Review
:wave:  @mrnugget @drewvanstone @br-lewis @makkes  Can you review, please?

File: packages/dcos-integration-test/extra/test_dcos_diagnostics.py [hunk 1]
```diff
@@ -479,6 +479,9 @@ def _download_bundle_from_master(dcos_api_session, master_index, bundle):
         'var/lib/dcos/exhibitor/zookeeper/snapshot/myid.gz',
         'var/lib/dcos/exhibitor/conf/zoo.cfg.gz',
         'var/lib/dcos/mesos/log/mesos-master.log.gz',
+        'var/lib/dcos/mesos/log/mesos-master.log.1.gz',
+        'var/lib/dcos/mesos/log/mesos-master.log.2.gz.gz',
```
Comment (drewvanstone) [inline packages/dcos-integration-test/extra/test_dcos_diagnostics.py]: Did you mean for this to be `*.gz.gz`?

File: gen/dcos-config.yaml [hunk 2]
```diff
@@ -1765,7 +1765,23 @@ package:
               },
               {
                   "Location": "{{ mesos_master_log_file }}",
-                  "Role": ["master"]
+                  "Role": ["master"],
+                  "Optional": true
+              },
+              {
+                  "Location": "{{ mesos_master_log_file }}.1",
```
Comment (drewvanstone) [inline gen/dcos-config.yaml]: Should this be `*.1.gz`, or are we expecting it not to be gzipped?

File: packages/dcos-integration-test/extra/test_dcos_diagnostics.py [hunk 3]
```diff
@@ -479,6 +479,9 @@ def _download_bundle_from_master(dcos_api_session, master_index, bundle):
         'var/lib/dcos/exhibitor/zookeeper/snapshot/myid.gz',
         'var/lib/dcos/exhibitor/conf/zoo.cfg.gz',
         'var/lib/dcos/mesos/log/mesos-master.log.gz',
+        'var/lib/dcos/mesos/log/mesos-master.log.1.gz',
```
Comment (drewvanstone) [inline packages/dcos-integration-test/extra/test_dcos_diagnostics.py]: Seems like ether the dcos-config.yaml:1772 needs to change, or this should drop the `.gz`.

Comment (janisz) [reply packages/dcos-integration-test/extra/test_dcos_diagnostics.py]: dcos-diagnostics compress file content so the bundled file ends with `.gz` suffix that's why we have `.gz.gz`  below

File: gen/dcos-config.yaml (hunk 2 above)
Comment (janisz) [reply gen/dcos-config.yaml]: I took this list from this comment https://github.com/dcos/dcos/pull/3589#discussion_r223855676
@kaysoky Can you confirm it's OK?

**Comment by janisz** [comment]
:wave:  @kaysoky @drewvanstone @br-lewis @makkes  Can you review, please?

File: gen/dcos-config.yaml (hunk 2 above)
Comment (kaysoky) [reply gen/dcos-config.yaml]: Yes, we use the `delaycompress` option of logrotate, which means the first rotated file will not be compressed.

https://github.com/dcos/dcos/blob/226f490f6769986a832c8fc8613c71c9cd7f45ac/gen/dcos-config.yaml#L1256

Comment (kaysoky) [reply gen/dcos-config.yaml]: Listing out these files probably is not the ideal way to grab the logs.  The number of logs is configurable, and by default, goes up to `*.136.gz`.

https://github.com/dcos/dcos/blob/226f490f6769986a832c8fc8613c71c9cd7f45ac/gen/calc.py#L213-L218

Comment (janisz) [reply gen/dcos-config.yaml]: Agree but we won't include them all in diagnostics bundle. How much of them do you need?

Comment (kaysoky) [reply gen/dcos-config.yaml]: If it is impractical to grab everything, then the existing amount (4 files, 0.75 - 1 GB total) is a reasonable middle ground, assuming diagnostic bundles are grabbed shortly after spotting a problem/reporting.

**Pull request merged**